### PR TITLE
added demoteTeam hook call

### PR DESCRIPTION
**Pull request by fruitwasp**
works just like [lockpickTime](http://wiki.darkrp.com/index.php/Hooks/Shared/lockpickTime)

### CODE REVIEW

File: gamemode/modules/base/sv_gamemode_functions.lua [hunk 1]
```diff
@@ -705,7 +705,9 @@ function GM:PlayerSpawn(ply)
 
     if ply.demotedWhileDead then
         ply.demotedWhileDead = nil
-        ply:changeTeam(GAMEMODE.DefaultTeam, true)
+        
+        local demoteTeam = hook.Call( "demoteTeam", nil, target ) or GAMEMODE.DefaultTeam
```
Comment (FPtje) [inline gamemode/modules/base/sv_gamemode_functions.lua]: Please don't use spaces between parentheses.

File: gamemode/modules/jobs/sv_jobs.lua [hunk 2]
```diff
@@ -259,7 +259,8 @@ local function FinishDemote(vote, choice)
     if choice == 1 then
         target:teamBan()
         if target:Alive() then
-            target:changeTeam(GAMEMODE.DefaultTeam, true)
+            local demoteTeam = hook.Call( "demoteTeam", nil, target ) or GAMEMODE.DefaultTeam
```
Comment (FPtje) [inline gamemode/modules/jobs/sv_jobs.lua]: Same here.

File: gamemode/modules/base/sv_gamemode_functions.lua (hunk 1 above)
Comment (FPtje) [inline gamemode/modules/base/sv_gamemode_functions.lua]: Also, there's an error. It's not supposed to be `target`, but `ply`. I won't merge this until you tested this. Test everything, even the small changes.

**Comment by fruitwasp** [comment]
fixed

**Comment by FPtje** [comment]
did you test it?

**Comment by fruitwasp** [comment]
yes, hon.

**Comment by FPtje** [comment]
I'll take your word for it.

**Pull request merged**